src/Module/Settings/Uexport.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * @file src/Modules/Settings/Uexport.php
+ */
+
+namespace Friendica\Module\Settings;
+
+use Friendica\App;
+use Friendica\App\Arguments;
+use Friendica\BaseModule;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\System;
+use Friendica\Database\DBA;
+use Friendica\Database\DBStructure;
+use Friendica\Module\BaseSettingsModule;
+
+/**
+ * Module to export user data
+ **/
+class Uexport extends BaseSettingsModule
+{
+	/**
+	 * Handle the request to export data.
+	 * At the moment one can export two different data set
+	 * 1. The profile data that can be used by uimport to resettle
+	 *    to a different Friendica instance
+	 * 2. The entire data-set, profile plus postings
+	 *
+	 * If there is an action required through the URL / path, react
+	 * accordingly and export the requested data.
+	 **/
+	public static function content()
+	{
+		parent::content();
+		$args = self::getClass(Arguments::class);
+		if ($args->getArgc() == 3) {
+			// @TODO Replace with router-provided arguments
+			$action = $args->get(2);
+			header("Content-type: application/json");
+			header('Content-Disposition: attachment; filename="' . $a->user['nickname'] . '.' . $action . '"');
+			switch ($action) {
+				case "backup":
+					SELF::uexport_all($a);
+					exit();
+					break;
+				case "account":
+					SELF::uexport_account($a);
+					exit();
+					break;
+				default:
+					exit();
+			}
+		}
+
+		/**
+		 * options shown on "Export personal data" page
+		 * list of array( 'link url', 'link text', 'help text' )
+		 */
+		$options = [
+			['settings/uexport/account', L10n::t('Export account'), L10n::t('Export your account info and contacts. Use this to make a backup of your account and/or to move it to another server.')],
+			['settings/uexport/backup', L10n::t('Export all'), L10n::t("Export your accout info, contacts and all your items as json. Could be a very big file, and could take a lot of time. Use this to make a full backup of your account \x28photos are not exported\x29")],
+		];
+		Hook::callAll('uexport_options', $options);
+
+		$tpl = Renderer::getMarkupTemplate("uexport.tpl");
+		return Renderer::replaceMacros($tpl, [
+			'$title' => L10n::t('Export personal data'),
+			'$options' => $options
+		]);
+	}
+	private function uexport_multirow($query) {
```

Comment:
should be `private static function` because this class is stateless (like `public static function content`).

Stateful: `$this->` operator
Stateless: `self::` operator

Codestyle: plz use Camel case at method names (`exportMultirow()`)

And plz add typehints to the method.

I think this would fit: `private static function exportMultirow(string $query)`